pkg/processing/find_field.go

```diff
@@ -99,54 +98,64 @@ func FindRangesFromIndexList(stack *nodestack.NodeStack, indexList []string, vm
+		if len(foundFields) == 0 {
 			return nil, fmt.Errorf("field %s was not found in ast.DesugaredObject", index)
 		}
 		if len(indexList) == 0 {
-			return []objectRange{fieldToRange(foundField)}, nil
-		}
-		switch fieldNode := foundField.Body.(type) {
-		case *ast.Var:
-			bind := FindBindByIdViaStack(stack, fieldNode.Id)
-			if bind == nil {
-				return nil, fmt.Errorf("could not find bind for %s", fieldNode.Id)
+			for i, found := range foundFields {
+				if i == 0 || foundFields[i-1].PlusSuper {
+					ranges = append(ranges, fieldToRange(found))
+				}
 			}
-			foundDesugaredObjects = append(foundDesugaredObjects, bind.Body.(*ast.DesugaredObject))
-		case *ast.DesugaredObject:
-			stack = stack.Push(fieldNode)
-			foundDesugaredObjects = append(foundDesugaredObjects, findDesugaredObjectFromStack(stack))
-		case *ast.Index:
-			tempStack := nodestack.NewNodeStack(fieldNode)
-			additionalIndexList := tempStack.BuildIndexList()
-			additionalIndexList = append(additionalIndexList, indexList...)
-			result, err := FindRangesFromIndexList(stack, additionalIndexList, vm)
-			if sameFileOnly && len(result) > 0 && result[0].Filename != stack.From.Loc().FileName {
-				continue
+			return ranges, nil
+		}
+
+		for _, foundField := range foundFields {
+			switch fieldNode := foundField.Body.(type) {
+			case *ast.Var:
+				bind := FindBindByIdViaStack(stack, fieldNode.Id)
+				if bind == nil {
+					return nil, fmt.Errorf("could not find bind for %s", fieldNode.Id)
+				}
+				foundDesugaredObjects = append(foundDesugaredObjects, bind.Body.(*ast.DesugaredObject))
+			case *ast.DesugaredObject:
+				stack = stack.Push(fieldNode)
+				foundDesugaredObjects = append(foundDesugaredObjects, findDesugaredObjectFromStack(stack))
+			case *ast.Index:
+				tempStack := nodestack.NewNodeStack(fieldNode)
+				additionalIndexList := tempStack.BuildIndexList()
+				additionalIndexList = append(additionalIndexList, indexList...)
+				result, err := FindRangesFromIndexList(stack, additionalIndexList, vm)
+				if sameFileOnly && len(result) > 0 && result[0].Filename != stack.From.Loc().FileName {
+					continue
+				}
+				return result, err
+			case *ast.Import:
+				filename := fieldNode.File.Value
+				rootNode, _, _ := vm.ImportAST(string(fieldNode.Loc().File.DiagnosticFileName), filename)
+				foundDesugaredObjects = findTopLevelObjects(nodestack.NewNodeStack(rootNode), vm)
 			}
-			return result, err
-		case *ast.Import:
-			filename := fieldNode.File.Value
-			rootNode, _, _ := vm.ImportAST(string(fieldNode.Loc().File.DiagnosticFileName), filename)
-			foundDesugaredObjects = findTopLevelObjects(nodestack.NewNodeStack(rootNode), vm)
 		}
 	}
 
-	return []objectRange{fieldToRange(foundField)}, nil
+	return ranges, nil
 }
 
-func findObjectFieldInObjects(objectNodes []*ast.DesugaredObject, index string) *ast.DesugaredObjectField {
+func findObjectFieldInObjects(objectNodes []*ast.DesugaredObject, index string) []*ast.DesugaredObjectField {
```

Comment:
I think we should change `findObjectFieldInObjects` to `findObjectFieldsInObjects` since now we're returning a slice.